plugins/common/lambda-rules-registration.hpp

```diff
@@ -0,0 +1,131 @@
+#ifndef LAMBDARULESREGISTRATION_HPP
+#define LAMBDARULESREGISTRATION_HPP
+
+#include <map>
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "wayfire/object.hpp"
+#include "wayfire/nonstd/observer_ptr.h"
+#include "wayfire/rule/lambda_rule.hpp"
+
+namespace wf
+{
+
+struct lambda_rule_registration_t;
+
+using map_type = std::map<std::string, std::shared_ptr<lambda_rule_registration_t>>;
+
+/**
+ * @brief The lambda_rule_registration_t struct represents registration information for a single lambda rule.
```

Comment:
I would like to also see how the rule text is used.

Reply:
Done